### ISSUE TITLE
[fix] Improve screen reader experience of Validation

### ISSUE THREAD

**Issue by kleinschmidtj** [trussworks/react-uswds#2510]
**ReactUSWDS Version & USWDS Version:**

ReactUSWDS 5.1.1

**Describe the bug**

The Validation component does not read out for screen readers updates to the status of the validation items once they are completed or not.

**To Reproduce**

1. Go to the validation component storybook: https://trussworks.github.io/react-uswds/?path=/story/components-validation--default
2. Open VoiceOver or another screen reader
3. Enter text into the input including a capital character
4. The checkmark appears but the screen reader doesn't read anything out

**Expected behavior**

The screen reader announces the status of the validation items in the list as "complete" or "incomplete" just as it does for the USWDS component: https://designsystem.digital.gov/components/validation/

**Screenshots**

**Additional context**

This PR adds generated aria-labels for each checklist item: https://github.com/uswds/uswds/pull/4914

This was added in USWDS 3.2.0

**Device and Browser Information (please complete the following information if describing a UI bug):**
